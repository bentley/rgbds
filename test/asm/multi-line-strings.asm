S EQUS "Hello"

PRINTT "\"\"\"\n"

PRINTT """{S}
world
"""

PRINTT """The multi-line string \ ; line continuations work
can contain:
- "single quotes"
- ""double quotes""
- even escaped \"""triple"\"" ""\"quotes\"\"\"
!"""

PRINTT """\n"""

printarg: MACRO
	PRINTT "arg <\1>\n"
	PRINTT """arg (\1)\n"""
ENDM

	printarg "
	printarg """

EMPTY1 EQUS ""
EMPTY2 EQUS "\ ; comment
"
EMPTY3 EQUS """"""
EMPTY4 EQUS """\ ; comment
"""
	PRINTT STRCAT("(", "{EMPTY1}", "{EMPTY2}", "{EMPTY3}", "{EMPTY4}", ")\n")
