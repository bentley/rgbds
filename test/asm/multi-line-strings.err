warning: multi-line-strings.asm(34): [-Wuser]
    check the line number
