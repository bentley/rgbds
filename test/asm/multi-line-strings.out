"""
Hello
world
The multi-line string can contain:
- "single quotes"
- ""double quotes""
- even escaped """triple""" """quotes"""
!
arg <">
arg (")
arg <""">
arg (""")
()
